src/plugins/index_pattern_management/server/routes/resolve_index.ts

```diff
@@ -60,15 +60,26 @@ export function registerResolveIndexRoute(router: IRouter): void {
 
       const dataSourceId = req.query.data_source;
       if (dataSourceId) {
-        const result = await (
-          await context.dataSource.opensearch.getClient(dataSourceId)
-        ).indices.resolveIndex({
-          name: encodeURIComponent(req.params.query),
-          expand_wildcards: req.query.expand_wildcards,
-        });
-        return res.ok({ body: result.body });
+        try {
+          const result = await (
+            await context.dataSource.opensearch.getClient(dataSourceId)
+          ).indices.resolveIndex({
+            name: encodeURIComponent(req.params.query),
+            expand_wildcards: req.query.expand_wildcards,
+          });
+          return res.ok({ body: result.body });
+        } catch (err: any) {
+          return res.customError({
```

Comment:
is this conversion needed vs directly throw err?

Reply:
directly throw error will cause 500

Reply:
@zengyan-amazon 
Because of `router.ts` will convert any error that's not a Client 401, to internal server error
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/c18736a07388d600a8b23561e1291495a081acb4/src/core/server/http/router/router.ts#L291-L303

Actually I am thinking I should add another `if..else` to the `router.ts` for data source errors, and cast it to a 400

Reply:
that will also apply all the places that consumes the client, but not having the error handling logic, fyi @noCharger